stripe/AndroidManifest.xml

```diff
@@ -19,6 +19,27 @@
         <activity
             android:name=".view.PaymentRelayActivity"
             android:theme="@style/StripeTransparentTheme" />
+
+        <!--
+        Set android:launchMode="singleTop" so that the StripeBrowserLauncherActivity instance that
+        launched the browser Activity will also handle the return URL deep link.
+        -->
+        <activity
+            android:name=".payments.StripeBrowserLauncherActivity"
+            android:theme="@style/StripeTransparentTheme"
+            android:launchMode="singleTop">
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <!--
+                Must match the value used in `ConfirmStripeIntentParams.DEFAULT_RETURN_URL`.
+                -->
+                <data android:scheme="stripesdk" android:host="payment_return_url" />
+            </intent-filter>
```

Comment:
hmm sorry I'm a bit confused with this change, when is the intent-filter triggered? IIUC it's used when some other app tries to start an activity with a `View` Intent that matches the scheme `stripesdk://payment_return_url`, but it looks like we didn't expose that url to public, nor did we use the scheme to start an intent internally inside our sdk

We do programmatically select this activity in `PaymentAuthWebViewContract` when `shouldUseBrowser` is true, but that would not require an intent-filter right?

Reply:
The intent-filter is triggered when a matching URL is opened by the final web page of the 3DS1 authentication flow (i.e. the SDK doesn't directly open this URL).

This URL will be set as the default return_url value when the user hasn't set a value themselves. It's intentionally not public at this point.

The value is defined here:
https://github.com/stripe/stripe-android/blob/323fc301aec2e833dd4233079867e092599d6fd0/stripe/src/main/java/com/stripe/android/model/ConfirmStripeIntentParams.kt#L30